### PR TITLE
Yitong/grafana deploy

### PR DESCRIPTION
**Pull request by YitongFeng**
Add grafana deployment.
Test done on eastus cluster.

### CODE REVIEW

File: service-deployment/clusterconfig-example.yaml [hunk 1]
```diff
@@ -85,6 +85,18 @@ clusterinfo:
     cluster_status_uri: http://10.0.3.9:9090
     # port for webportal
     server_port: 9286
+  
+  prometheusinfo:
+    # The address of the prometheus to connect to
+    prometheus_url: http://10.0.3.9:9090
+    # port for prometheus port
+    prometheus_port: 9090
```
Comment (fanyangCS) [inline service-deployment/clusterconfig-example.yaml]: you don't need this. prometheus deployment handles this, right?

File: service-deployment/bootstrap/grafana/cleanup.sh.template [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+# BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+# DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+pushd $(dirname "$0") > /dev/null
+
+kubectl delete deployment grafana -n monitor
```
Comment (ydye) [inline service-deployment/bootstrap/grafana/cleanup.sh.template]: Actually I don't think the monitor module should be in another namespace. @fanyangCS @YanjieGao

File: service-deployment/clusterconfig-example.yaml [hunk 3]
```diff
@@ -85,6 +85,18 @@ clusterinfo:
...
+  
+  grafanainfo:  
+    # The address of the grafana to connect to
+    grafana_url: http://10.0.3.9:3000
```
Comment (hwuu) [inline service-deployment/clusterconfig-example.yaml]: Why do you set the port info twice here? Duplicated configuration?

File: service-deployment/bootstrap/grafana/grafana.yaml.template [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        imagePullPolicy: Always
+        ports:
+        - containerPort: {{ clusterinfo['grafanainfo']['grafana_port'] }}
+          protocol: TCP        
+        env:
+        - name : GRAFANA_URL
+          value: {{ clusterinfo['grafanainfo']['grafana_url'] }}
+          # The following env variables are required to make Grafana accessible via
+          # the kubernetes api-server proxy. On production clusters, we recommend
+          # removing these env variables, setup auth for grafana, and expose the grafana
```
Comment (ydye) [inline service-deployment/bootstrap/grafana/grafana.yaml.template]: The grafana in PAI is accessed by IP or kubernetes service?

File: service-deployment/service.yaml [hunk 5]
```diff
@@ -162,4 +171,14 @@ servicelist:
       - webportal-ds.yaml
       - cleanup.sh
     startscript: start.sh
+    stopscript: cleanup.sh
+
+  grafana:
+    prerequisite:
```
Comment (hwuu) [inline service-deployment/service.yaml]: Should be `prerequisite: None`?

File: service-deployment/clusterconfig-example.yaml (hunk 1 above)
Comment (YitongFeng) [reply service-deployment/clusterconfig-example.yaml]: It is used to replace the prom-datasource.yaml.template when building the grafana docker. I didn't include this file in the previous PR.

File: service-deployment/service.yaml (hunk 5 above)
Comment (YitongFeng) [reply service-deployment/service.yaml]: Yes, set it at line 178

File: service-deployment/bootstrap/grafana/grafana.yaml.template (hunk 4 above)
Comment (YitongFeng) [reply service-deployment/bootstrap/grafana/grafana.yaml.template]: we can access by IP and don't need service, I have removed the grafana service .

File: service-deployment/bootstrap/grafana/cleanup.sh.template (hunk 2 above)
Comment (YitongFeng) [reply service-deployment/bootstrap/grafana/cleanup.sh.template]: Solved, removed namespace.

File: service-deployment/clusterconfig-example.yaml (hunk 3 above)
Comment (YitongFeng) [reply service-deployment/clusterconfig-example.yaml]: Solved

**Pull request merged**